enable type check common_distributed

alanadakotashine opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44812 enable type check common_distributed**

Differential Revision: [D23747466](https://our.internmc.facebook.com/intern/diff/D23747466)

Commit:
enable type check common_distributed

[ghstack-poisoned]

Commit:
Update on "enable type check common_distributed"




[ghstack-poisoned]

Commit:
Update on "enable type check common_distributed"




[ghstack-poisoned]

walterddr commented on mypy.ini:
```diff
@@ -65,8 +65,8 @@ ignore_errors = True
 
-[mypy-torch.testing._internal.common_distributed.*]
-ignore_errors = True
+#[mypy-torch.testing._internal.common_distributed.*]
+#ignore_errors = True
```
can you delete the lines entirely?

walterddr reviewed (changes requested):
overall looks good. just one comment, please fix it and reimport to phab. thanks --Rong

Commit:
Update on "enable type check common_distributed"


Differential Revision: [D23747466](https://our.internmc.facebook.com/intern/diff/D23747466)

[ghstack-poisoned]

walterddr reviewed (approved):
lgtm. please re-import to phab
